src/main/java/org/openRealmOfStars/starMap/planet/Planet.java

```diff
@@ -467,36 +467,34 @@ public void removeBuilding(Building building) {
    * @return The production time
    */
   public String getProductionTime(Construction build) {
-    int metalReq = build.getMetalCost()-getMetal();
-    int prodReq = build.getProdCost()-getProdResource();
-    int metalTurn = 0;
-    int prodTurn = 0;
-    if (metalReq <= 0 &&prodReq <= 0) {
+    int metalReq = build.getMetalCost() - getMetal();
+    int prodReq = build.getProdCost() - getProdResource();
+    if (metalReq <= 0 && prodReq <= 0) {
       return "1 turn";
     }
-    if (getTotalProduction(PRODUCTION_METAL) > 0 && metalReq > 0) {
-      metalTurn = (int) Math.ceil((double) metalReq / (double) getTotalProduction(PRODUCTION_METAL));
-    } else if (getTotalProduction(PRODUCTION_METAL) == 0 && metalReq > 0) {
-      metalTurn = -1;
-    } else if (metalReq <= 0) {
-      metalTurn = 1;
-    }
-    if (getTotalProduction(PRODUCTION_PRODUCTION) > 0 && prodReq > 0) {
-      prodTurn = (int) Math.ceil((double) prodReq / (double) getTotalProduction(PRODUCTION_PRODUCTION));
-    } else if (getTotalProduction(PRODUCTION_PRODUCTION) == 0 && prodReq > 0) {
-      prodTurn = -1;
-    } else if (prodReq <= 0) {
-      prodTurn = 1;
-    }
-    if (prodTurn == -1 || metalTurn== -1) {
+    int metalTurn = getProductionTimeByProductionType(metalReq, PRODUCTION_METAL);
+    int prodTurn = getProductionTimeByProductionType(prodReq, PRODUCTION_PRODUCTION);
+    if (prodTurn == -1 || metalTurn == -1) {
       return "Never";
     } 
     if (prodTurn > metalTurn) {
-      return prodTurn+" turns";
+      return prodTurn + " turns";
     }
-    return metalTurn+" turns";
+    return metalTurn + " turns";
   }
-  
+
+  private int getProductionTimeByProductionType(int productionReq, int productionType) {
```

Comment:
There should be javadocs.
/**
* Get production time in turn depending on production type.
* This method estimates how long construction production takes in turn.
* This can be used for both metal and production estimations.
* @param productionReq How much production is required
* @param productionType Either PRODUCTION_PRODUCTION or PRODUCTION_METAL